Blog Post: Custom View Layout (#655)
home-assistant/developers.home-assistant: blog/2020-09-30-lazyMoreInfo-viewChanges.md

---
author: Zack Barett
authorURL: https://github.com/zsarnett
authorTwitter: zsarnett
title: "Lovelace: Lazy Loading More Info and Custom View Layouts"
---

Custom Element developers can now create Custom View Layouts that users can load and use!

In 0.116, we will be changing the way that we create Views in Lovelace. In the past, we had 2 views, a default view, and a panel view. When talking about adding Drag and Drop to Lovelace we decided we could do even better and start allowing custom view types.

Custom Developers will now be able to create a view that receives the following properties:

```ts
interface LovelaceViewElement {
  hass?: HomeAssistant;
  lovelace?: Lovelace;
  index?: number;
  cards?: Array<LovelaceCard | HuiErrorCard>;
  badges?: LovelaceBadge[];
  setConfig(config: LovelaceViewConfig): void;
}
```

> The cards and badges will be already created and HASS will be updated on each card so that you do not need to worry about that in your code.

Author: ```suggestion
The cards and badges will be already created and Hass will be updated on each card so that you do not need to worry about updating Hass on the Cards array for every Hass update.
```

> The cards and badges will be already created and Hass will be updated on each card so that you do not need to worry about updating Hass on the Cards array for every Hass update.

Author: ```suggestion
Cards and Badges will be created and maintained by the core code and given to the custom view. The custom views are meant to load the cards and badges and display them in a customized layout.
```

Author: ```suggestion
Cards and Badges will be created and maintained by the core code and given to the custom view. The custom views are meant to load the cards and badges and display them in a customized layout.
```

> Here is an example below: (note: this example does not have all of the properties but necessities to show the example)

Author: ```suggestion
Here is an example below: (note: this example does not have all of the properties but the necessities to show the example)
```

---
author: Zack Barett
authorURL: https://github.com/zsarnett
authorTwitter: zsarnett
title: "Lovelace: Lazy Loading More Info and Custom View Layouts"
---

Custom Element developers can now create Custom View Layouts that users can load and use!

In 0.116, we will be changing the way that we create Views in Lovelace. In the past, we had 2 views, a default view, and a panel view. When talking about adding Drag and Drop to Lovelace we decided we could do even better and start allowing custom view types.

Custom Developers will now be able to create a view that receives the following properties:

```ts
interface LovelaceViewElement {
  hass?: HomeAssistant;
  lovelace?: Lovelace;
  index?: number;
  cards?: Array<LovelaceCard | HuiErrorCard>;
  badges?: LovelaceBadge[];
  setConfig(config: LovelaceViewConfig): void;
}
```

The cards and badges will be already created and Hass will be updated on each card so that you do not need to worry about updating Hass on the Cards array for every Hass update.

Here is an example below: (note: this example does not have all of the properties but the necessities to show the example)

> ```ts

Reviewer: Let's make the example in JS so people can copy paste it and use it ?

---
author: Zack Barett
authorURL: https://github.com/zsarnett
authorTwitter: zsarnett
title: "Lovelace: Custom View Layouts"
---

Custom Element developers can now create Custom View Layouts that users can load and use!

In 0.116, we will be changing the way that we create Views in Lovelace. In the past, we had 2 views, a default view, and a panel view. When talking about adding Drag and Drop to Lovelace we decided we could do even better and start allowing custom view types.

Custom Developers will now be able to create a view that receives the following properties:

```ts
interface LovelaceViewElement {
  hass?: HomeAssistant;
  lovelace?: Lovelace;
  index?: number;
  cards?: Array<LovelaceCard | HuiErrorCard>;
  badges?: LovelaceBadge[];
  setConfig(config: LovelaceViewConfig): void;
}
```

Cards and Badges will be created and maintained by the core code and given to the custom view. The custom views are meant to load the cards and badges and display them in a customized layout.

Here is an example below: (note: this example does not have all of the properties but the necessities to show the example)

```js
class MyNewView extends LitElement {

>   public cards = [];
> 
>   public setConfig(_config) {}

Author: ```suggestion
  cards = [];

  setConfig(_config) {}
```
